### PR TITLE
 Added Json Library.

### PR DESCRIPTION
**Pull request by bythos14**
Let's try again.

### CODE REVIEW

File: docs/Doxyfile [hunk 1]
```diff
@@ -2480,4 +2480,4 @@ GENERATE_LEGEND        = YES
 # The default value is: YES.
 # This tag requires that the tag HAVE_DOT is set to YES.
 
-DOT_CLEANUP            = YES
+DOT_CLEANUP            = YES
```
Comment (d3m3vilurr) [inline docs/Doxyfile:2483]: could you keep the last return? :)

File: include/psp2/json.h [hunk 2]
```diff
@@ -0,0 +1,897 @@
...
+	SCE_JSON_VALUE_OBJECT	= 7  //!< Object value
+} ValueType;
+
+/**
+ * Enumerator of different errors in this library.
+ */
+typedef enum SceJsonError
+{
+	SCE_JSON_ERROR_UNINIT						= 0x80920110, //!< The module has not been initialised. 
+	SCE_JSON_ERROR_REINIT						= 0x80920111, //!< The module has already been initialised.
```
Comment (d3m3vilurr) [inline include/psp2/json.h:39]: how about `ERROR_UNLOADED`, `ERROR_LOADED`?

File: include/psp2/json.h [hunk 3]
```diff
@@ -0,0 +1,897 @@
...
+	SCE_JSON_ERROR_MEM_FAIL						= 0x80920102, //!< Internal Memory Allocation failure(returned nullptr).
```
Comment (d3m3vilurr) [inline include/psp2/json.h:40]: `ERROR_MEM_ALLOC`?

File: include/psp2/json.h [hunk 4]
```diff
@@ -0,0 +1,897 @@
...
+	SCE_JSON_PARSER_ERROR_EMPTY_BUF				= 0x80920105, //!< InitParameter.unk_0x8 is 0.
+	SCE_JSON_PARSER_ERROR_FILE_LOAD_FAIL		= 0x80920103, //!< Parser failed to load file.
+	SCE_JSON_PARSER_ERROR_BAD_CHAR				= 0x80920101, //!< Invalid character in the parsed data.
```
Comment (d3m3vilurr) [inline include/psp2/json.h:43]: `ERROR_INVALID_TOKEN`?

**Comment by bythos14** [comment]
Yes, I do have it working in a homebrew. I'll have to make those changes tomorrow, would you like me to squash them into the same commit as adding the header file or is a separate commit okay?

**Comment by d3m3vilurr** [comment]
well, i like to rebase to one header commit and one  doxygen commit. :)

**Comment by joel16** [comment]
There seems to be a lot of indentation issues in your json header.

File: include/psp2/json.h [hunk 5]
```diff
@@ -0,0 +1,897 @@
...
+	/**
+	 * Serializes the data into a string in json format
+	 * 
+	 * @param[out] s - The resulting string, can be written to a file or parsed.
+	 * 
+	 * @return 0 on success, <0 on error.
+	 */
+	int			serialize(String& s);
+	/**
+	 * Serializes the values into a string in json format. Allows for a callback for the different stages in the serialization.
```
Comment (d3m3vilurr) [inline include/psp2/json.h:655]: please add line breaks to keep the 80cols.

**Comment by bythos14** [comment]
> There seems to be a lot of indentation issues in your json header.

That seems to be due to github having tabs sized at around 2 or 3 spaces, I did indentation with tabs sized at 4 spaces so it does look a tad off on the site.

**Comment by bythos14** [comment]
Also, I made the changes you asked about

**Comment by d3m3vilurr** [comment]
8 space tabstop is ok. imo @joel16 told, method definitions has tab char in middle.
```
	void		setAllocatorInfoCallBack(AllocatorInfoCallback cb, void* unk0);
```
should be
```
	void setAllocatorInfoCallBack(AllocatorInfoCallback cb, void* unk0);
```

**Comment by bythos14** [comment]
Fixed the indentation issues

**Pull request merged**